projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/SupportedPythonVersions.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import java.security.InvalidParameterException;
+import java.util.*;
+
+import org.springframework.beans.factory.annotation.Value;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Handles operations related to supported python versions for data job deployments. The class
+ * contains utility methods which provide functionality to read the configuration related to the
+ * python versions supported by the Control Service. These utility methods are meant to be used in
+ * other components, as needed.
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SupportedPythonVersions {
+
+  public record DeploymentDetails(String baseImage, String vdkImage) {}
+
+  @Value("#{${datajobs.deployment.supportedPythonVersions:{}}}")
+  private Map<String, DeploymentDetails> supportedPythonVersions;
+
+  @Value("${datajobs.deployment.defaultPythonVersion}")
+  private String defaultPythonVersion;
+
+  /**
+   * Check if the pythonVersion passed by the user is supported by the Control Service.
+   *
+   * @param pythonVersion python version passed by the user.
+   * @return true if the version is supported, and false otherwise.
+   */
+  public boolean isPythonVersionSupported(String pythonVersion) {
+    return supportedPythonVersions.containsKey(pythonVersion);
+  }
+
+  /**
+   * Returns a set of the python versions supported by the Control Service, in the format: [3.7,
+   * 3.8, ...]. If the supportedPythonVersions configuration is not set, the method returns an empty
+   * set.
+   *
+   * @return A set of all python versions supported by the Control Service.
+   */
+  public Set<String> getSupportedPythonVersions() {
+    return Optional.ofNullable(supportedPythonVersions)
```

Comment:
You can just return the keySet since supportedPythonVersions has a default value.
```suggestion
    return supportedPythonVersions.keySet();
```

Reply:
Fixed, thanks